app/controllers/maintenance_tasks/runs_controller.rb

```diff
@@ -34,6 +34,19 @@ def cancel
       redirect_to(task_path(@run.task_name), alert: error.message)
     end
 
+    # Downloads the CSV attachment associated with the Run.
+    def csv_file
+      respond_to do |format|
+        format.csv do
+          csv_file = @run.csv_file.download
+          task_name = @run.task_name.underscore.gsub('/', '_')
+          timestamp = @run.created_at.strftime('%Y%m%d%H%M%S')
+
+          send_data(csv_file, filename: "#{task_name}_#{timestamp}.csv")
```

Comment:
Maybe we could put the timestamp first for better ordering? OTOH this sorts by task then date, it's pretty cool too.
Why the underscore? spaces are fine in file names, no?

Also maybe we could just use the name of the file that was sent to us to make things simpler if a bit less useful?

Reply:
Timestamp vs task name first / underscores vs spaces are not things I feel strongly about, so happy to change any of these if you'd like.

I'm torn on using the original file name. It simplifies things for sure, but OTOH it's kind of nice to have the task + date in the file name, especially if you're downloading multiple CSVs for a Task.